src/IB/IBFEMethod.cpp

```diff
@@ -1362,8 +1362,12 @@ void IBFEMethod::endDataRedistribution(Pointer<PatchHierarchy<NDIM> > /*hierarch
             // TODO: d_current_time is actually nan here. Since we don't do
             // any sort of tagging based on the current time I think we can
             // just put in a bogus (nonnegative) value.
-            d_scratch_gridding_algorithm->regridAllFinerLevels(
-                d_scratch_hierarchy, 0, 0.0 /*d_current_time*/, tag_buffer);
+            int max_level = d_scratch_hierarchy->getFinestLevelNumber();
+            if (max_level == 0)
+                d_scratch_gridding_algorithm->makeCoarsestLevel(d_scratch_hierarchy, 0.0 /*d_current_time*/);
+            else
+                d_scratch_gridding_algorithm->regridAllFinerLevels(
+                    d_scratch_hierarchy, max_level - 1, 0.0 /*d_current_time*/, tag_buffer);
```

Comment:
@abarret:
> Does the fact that regridAllFinerLevels can only add one level to the hierarchy per call affect how things happen as levels appear and disappear?

You're not wrong but that's irrelevant - this patch is a lot dumber than perhaps you think it is. At the point this is called all relevant levels exist (we are regridding for the first time). The original bug was that we called `regridAllFinerLevels` from level 0, which regrids level 1 - this level doesn't exist in the scratch hierarchy if we have, e.g., three total levels.

`makeCoarsestLevel` is oddly named in this context, but according to SAMRAI's documentation if the level already exists it applies load balancing to level 0 - which is exactly what we want to do.

Reply:
@abarret even if you don't believe me all the tests pass which shows that I am right :)